pandas/tests/tseries/offsets/test_offsets_properties.py

```diff
@@ -97,40 +95,39 @@ def test_on_offset_implementations(dt, offset):
     assert offset.is_on_offset(dt) == (compare == dt)
 
 
-@pytest.mark.xfail(
-    reason="res_v2 below is incorrect, needs to use the "
-    "commented-out version with tz_localize.  "
-    "But with that fix in place, hypothesis then "
-    "has errors in timezone generation."
-)
 @given(gen_yqm_offset, gen_date_range)
 def test_apply_index_implementations(offset, rng):
     # offset.apply_index(dti)[i] should match dti[i] + offset
-    assume(offset.n != 0)  # TODO: test for that case separately
 
-    # rng = pd.date_range(start='1/1/2000', periods=100000, freq='T')
+    # TODO: test for that case separately
+    assume(offset.n != 0)
+
     ser = pd.Series(rng)
 
     res = rng + offset
     res_v2 = offset.apply_index(rng)
-    # res_v2 = offset.apply_index(rng.tz_localize(None)).tz_localize(rng.tz)
+    res_v2 = offset.apply_index(rng.tz_localize(None)).tz_localize(rng.tz)
     assert (res == res_v2).all()
 
+    # apply_index is only for indexes, not series, so no res2_v2
     assert res[0] == rng[0] + offset
     assert res[-1] == rng[-1] + offset
     res2 = ser + offset
-    # apply_index is only for indexes, not series, so no res2_v2
+
+    # TODO: Check randomly assorted entries, not just first/last
```

Comment:
I am interested in doing that, anyone got a smart way of doing that?

Reply:
```
for _ in range(10):
     n = np.random.randint(len(ser))
     assert res2.iloc[n] == ser.iloc[n] + offset
```

(10 is arbitrary number here, you decide what is reasonable number